lib/action_view/component/callbacks.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require "active_support/concern"
+require "active_support/callbacks"
+
+module ActionView
+  module Component
+    module Callbacks
+      extend ActiveSupport::Concern
+      include ActiveSupport::Callbacks
+
+      included do
+        define_callbacks :render
+      end
+
+      module ClassMethods
+        def before_render(*filters, &block)
+          set_callback(:render, :before, *filters, &block)
+        end
```

Comment:
What does `filters` mean or get used for?

Reply:
with filters you can send method name instead of block
```ruby
  before_render :render?
  private def render?; end
```
https://github.com/rails/rails/blob/master/activejob/lib/active_job/callbacks.rb